templates/nextflow-ecs-service.yaml

```diff
@@ -0,0 +1,209 @@
+AWSTemplateFormatVersion: '2010-09-09'
+
+Description: Nextflow Tower ECS Service template
+
+Parameters:
+
+  ClusterName:
+    Type: String
+    Description: ECS cluster name
+
+  TaskDefinitionArn:
+    Type: String
+    Description: ARN for the ECS task definition that the ECS service uses
+
+  VpcId:
+    Description: ID of VPC
+    Type: AWS::EC2::VPC::Id
+
+  SubnetIds:
+    Type: CommaDelimitedList
+    Description: List of public Subnet Ids for the load balancer
+
+  SSLCertificateArn:
+    Description: 'ARN for a certificate that exists in the account and is valid for the domain'
+    Type: String
+
+  TowerUIContainerName:
+    Type: String
+    Description: Name of the tower ui container to associate with the load balancer
+
+  TowerUIContainerPort:
+    Type: Number
+    Description: The port on the tower ui container to associate with the load balancer
+
+  AlbListenerPort:
+    Type: Number
+    Description: (Optional) The port for the ALB HTTPS listener
+    Default: 443
+
+  SecurityIngressCidrIp:
+    Type: String
+    Description: >
+      Optional - Specifies the CIDR/IP range for Security Ports - defaults
+      to 0.0.0.0/0
+    AllowedPattern: ^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$
+    Default: '0.0.0.0/0'
+
+Resources:
+
+  EcsServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+              - ecs.amazonaws.com
+          Action:
+            - sts:AssumeRole
+      Path: /
+      Policies:
+      - PolicyName: ecs-service
+        PolicyDocument:
+          Statement:
+          - Effect: Allow
+            Action:
+              - elasticloadbalancing:DeregisterInstancesFromLoadBalancer
+              - elasticloadbalancing:DeregisterTargets
+              - elasticloadbalancing:Describe*
+              - elasticloadbalancing:RegisterInstancesWithLoadBalancer
+              - elasticloadbalancing:RegisterTargets
+              - ec2:Describe*
+              - ec2:AuthorizeSecurityGroupIngress
+            Resource: '*'
+
+  EcsAlbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupName: !Join
+        - '-'
+        - - !Ref 'AWS::StackName'
+          - EcsAlbSecurityGroup
+      GroupDescription: Security Group for Nextflow ECS load balancer
+      VpcId: !Ref VpcId
+
+  EcsAlbSecurityGroupIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !GetAtt EcsAlbSecurityGroup.GroupId
```

Comment:
There is an important difference b/tw `Ref` and `GetAtt`. Most CFN resources have some default value they return when you use the `Ref` function. For example, the security group documentation indicates that `Ref` returns the resource id rather than the group id. Many CFN resources allow one to access additional fields -- this is always through `GetAtt`. For a security group the additional fields are the GroupId and the VpcId.

See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html#aws-properties-ec2-security-group-return-values-ref

Reply:
Oh, I was under the impression that in this case, the resource ID returned by `Ref` would be the security group ID.